tracer/src/Datadog.Trace/Activity/Handlers/ActivityHandlersRegister.cs

```diff
@@ -9,12 +9,17 @@ namespace Datadog.Trace.Activity.Handlers
 {
     internal static class ActivityHandlersRegister
     {
+        // Disable Activity Handler does not listen to the activity source at all.
+        internal static readonly DisableActivityHandler DisableHandler = new();
```

Comment:
Is there any value in having a separate `DisableActivityHandler` vs `IgnoreActivityHandler` 🤔 Given that the `IgnoreActivityHandler` causes behaviour changes that it sounds like we don't want, would it make sense to _replace_ `IgnoreActivityHandler`?

i.e. should we allow people to disable handlers, but also change all our _existing_  "ignored" handlers to be disabled? 🤔

Reply:
So for the "Ignored" Activities we do go and update them for context propagation so they are ignored in the sense that we don't create a Datadog span for them, but we still do need to do things as we listen to the source. I think we want to keep that.

The "Disabled" toggles that listener off so we don't know anything about them.

Reply:
OK, cool, then I guess my question is: what are the implications if someone disables an activity that's in the "middle" of the trace? What about if it's a "root" activity? Will that mess up our tracing, or will we just have a gap (which is fine) but everything else in terms or parenting etc would work ok? I guess that's somewhat covered by the tests, but it's not entirely clear to me

Reply:
Those are good questions

I haven't really looked at this stuff in a long time so I don't know off the top of my head, but I will add test cases to the sample project to test this out.